UnityProject/Assets/Scripts/Core/Factories/EffectsFactory.cs

```diff
@@ -82,8 +87,8 @@ public static void BloodSplat(Vector3 worldPos, BloodSplatSize splatSize, BloodS
 			var matrix = MatrixManager.AtPoint(Vector3Int.RoundToInt(worldPos), true);
 			if (matrix.Matrix.Get<FloorDecal>(worldPos.ToLocalInt(matrix.Matrix), true).Count() == 0)
 			{
-				Spawn.ServerPrefab(chosenTile, worldPos,
-								   matrix.Objects);
+				Spawn.ServerPrefab(chosenTile, worldPos, matrix.Objects);
+				//TODO: Need to add blood reagents
```

Comment:
surely you just remove from player and add to splat?

Reply:
The blood splat doesn’t have pre-existing requisite‘s for adding reagents. Chem splats already had a method for it but it just wasn’t being used. 

Also I want to do the chemsplat‘s first and make sure that I wasn’t being naughty by using the chem splat game object to hold the reagent.

Lastly, I know that bleeding is pending in another PR and I don’t want to touch blood until that’s merged since it will just lead to conflicts. Everywhere where there is blood I will have to pass in a reagent mix.

Reply:
Oh and, The way the effects factory handled blood splats is a little rigid, because each blood size is it own prefab. I want to investigate using a better system where the blood splats are a single prefab but the sprite will reflect the current volume of reagents. The same goes for chemsplats (but right now we don’t have assets for many chemsplats)

Reply:
Sure, if/when you investigate that system, make sure you test that the sprites would be properly synced across all clients, as that'd likely be the catch to consolidating them.